src/test/app/LedgerLoad_test.cpp

```diff
@@ -0,0 +1,221 @@
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <test/support/jtx.h>
+#include <test/support/jtx/Env.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/SField.h>
+#include <boost/scope_exit.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string.hpp>
+#include <fstream>
+
+namespace ripple {
+
+class LedgerLoad_test : public beast::unit_test::suite
+{
+    Json::Value jvLedger_;
+    Json::Value jvHashes_;
+    boost::filesystem::path ledgerFile_;
+    boost::filesystem::path dbPath_;
+
+    auto ledgerConfig(std::string const& ledger, Config::StartUpType type)
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        p->START_LEDGER = ledger;
+        p->START_UP = type;
+        p->legacy("database_path", dbPath_.string());
+        return p;
+    }
+
+    // setup for test cases
+    void
+    setupLedger()
+    {
+        using namespace test::jtx;
+
+        boost::system::error_code ec;
+        // create a temporary path to write ledger files in
+        dbPath_  = boost::filesystem::temp_directory_path(ec);
+        BEAST_EXPECTS(!ec, ec.message());
+        dbPath_ /= boost::filesystem::unique_path("%%%%-%%%%-%%%%-%%%%", ec);
+        BEAST_EXPECTS(!ec, ec.message());
+        boost::filesystem::create_directories(dbPath_, ec);
+        BEAST_EXPECTS(!ec, ec.message());
+
+        ledgerFile_ = dbPath_ / "ledgerdata.json";
+
+        Env env {*this};
+
+        for(auto i = 0; i < 20; ++i)
+        {
+            Account acct {"A" + std::to_string(i)};
+            env.fund(XRP(10000), acct);
+            env.close();
+            if(i > 0)
+            {
+                Account prev {"A" + std::to_string(i-1)};
+                env.trust(acct["USD"](1000), prev);
+                env(pay(acct, prev, acct["USD"](5)));
+            }
+            env(offer(acct, XRP(100), acct["USD"](1)));
+            env.close();
+        }
+
+        jvLedger_ = env.rpc ("ledger", "current", "full") [jss::result];
+        BEAST_EXPECT(jvLedger_[jss::ledger][jss::accountState].size() > 0);
```

Comment:
had that originally and opted for a simpler check, but I'll go back to stricter checking.